src/boost_histogram/_internal/traits.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# This is basically a dataclass from Python 3.7, with frozen=True
+
+_traits = (
+    "underflow",
+    "overflow",
+    "circular",
+    "growth",
+    "continuous",
+    "inclusive",
+    "ordered",
+)
+
+
+# This can be converted to a immutable dataclass once Python < 3.7 is dropped.
+
+
+class Traits(object):
+    __slots__ = _traits
+
+    def __init__(
+        self,
+        underflow=False,
+        overflow=False,
+        circular=False,
+        growth=False,
+        continuous=False,
+        inclusive=False,
+        ordered=False,
+    ):
+        for name in _traits:
+            object.__setattr__(self, name, locals()[name])
```

Comment:
Why not `setattr(self, name, locals()[name]`?

Reply:
I think this was... I have no idea why I wrote it this way. Maybe a leftover from when I had some workarounds for Axis (which are no longer needed due to the `__dict__` in `__slots__` simplification), which for some reason I then put here? 🤦 Will clean up, thanks!